sympy/tensor/array/ndim_array.py

```diff
@@ -560,3 +562,15 @@ def as_immutable(self):
 
     def as_mutable(self):
         raise NotImplementedError("abstract method")
+
+
+@accept_shape.register(ArrayKind)
+def _(kind, expr):
+    return visit_shape_arraykind(expr)
```

Comment:
Rather than `visit_shape_arraykind` it would be better to call this something like `get_shape_array` because this function gets the shape of an array.